pandas/core/arrays/categorical.py

```diff
@@ -2947,9 +2947,11 @@ def recode_for_categories(
             return codes.copy()
         return codes
 
-    indexer = coerce_indexer_dtype(
-        new_categories.get_indexer(old_categories), new_categories
-    )
+    if new_categories._index_as_unique:
```

Comment:
You can use ``get_indexer_for`` which makes this decision for you

Reply:
nice. updated, thanks